modules/aleph/src/Aleph/Entity/AlephMaterial.php

```diff
@@ -322,9 +322,22 @@ public static function materialFromItem(\SimpleXMLElement $item) {
     $material->setTitle((string) $item->xpath('z13/z13-title')[0]);
     $material->setSubLibrary((string) $item->xpath('z30/z30-sub-library')[0]);
     $material->setCollection((string) $item->xpath('z30/z30-collection')[0]);
-    $material->setSubLibraryCode((string) $item->xpath('z36-sub-library-code')[0]);
-    $material->setDueDate((string) $item->xpath('z36/z36-due-date')[0]);
-    $material->setLoanDate((string) $item->xpath('z36/z36-loan-date')[0]);
+
+    if ($item->xpath('z30-sub-library-code')) {
+      $material->setSubLibraryCode((string) $item->xpath('z30-sub-library-code')[0]);
+    }
+
+    if ($item->xpath('z36-sub-library-code')) {
+      $material->setSubLibraryCode((string) $item->xpath('z36-sub-library-code')[0]);
+    }
```

Comment:
I would appreciate some documentation here if possible - preferably with a link to further documentation. It seems as if we can derive the sub-library code from two different elements in the response?